src/Menu/Matcher/Voter/ChildrenVoter.php

```diff
@@ -19,6 +19,7 @@
  * Children menu voter based on children items.
  *
  * @author Samusev Andrey <andrey.simfi@ya.ru>
+ * @deprecated this class is deprecated since 3.x and will be removed in 4.0
```

Comment:
you should also add a deprecation warning in the file.

And it is not deprecated since `3.x` (which is not a version number but a version range). It is deprecated since `3.28` (which will be the next 3.x release)

Reply:
@stof the 3.x is how we do things (we can't guarantee the PR will get merged in time for `3.28`, we deliberately choose to do it like this. As per the deprecation warning, @Soullivaneuh told me it wasn't required anymore.

Reply:
See https://github.com/sonata-project/dev-kit/pull/312

Reply:
@stof Does not the latest versions of Symfony handle single `@deprecated` tag on the profiler? I am sure it was working like that at a moment. :-|

Reply:
see my comment in the discussion linked above

Reply:
So this PR is good right? The point of it is to warn if someone extends the voter and tags it.

Reply:
Now that I think of it I can see how it could possibly work. Will add a deprecation.